server/flyway/sql/V46__add_spar_roles.sql

```diff
@@ -0,0 +1,53 @@
+-- Delete any role assignments to the SPAR initial testing roles and old SPAR roles
+DELETE FROM app_fam.fam_user_role_xref WHERE role_id IN
+    (SELECT role_id from app_fam.fam_role WHERE application_id IN
+        (SELECT application_id FROM app_fam.fam_application WHERE application_name IN ('SPAR_DEV', 'SPAR_TEST', 'SPAR_PROD'))
+    )
+;
+
+-- Delete previously created the SPAR initial testing roles and old SPAR roles for spar_dev, spar_test and spar_prod applications
+DELETE FROM app_fam.fam_role WHERE application_id IN (
+    SELECT application_id FROM app_fam.fam_application WHERE application_name IN ('SPAR_DEV', 'SPAR_TEST', 'SPAR_PROD')
+);
+
+-- Add roles for SPAR_DEV
+INSERT INTO app_fam.fam_role (
+    role_name,
+    role_purpose,
+    application_id,
+    role_type_code,
+    create_user,
+    create_date
+)
+VALUES ('SPAR_TSC_ADMIN', 'Allow ministry users to access SPAR administrative screens and perform administrative functions such as seedlot registration approval, or seedling request approval', (select application_id from app_fam.fam_application where application_name = 'SPAR_DEV'), 'C', CURRENT_USER, CURRENT_DATE),
```

Comment:
Maybe we can utilize the `with` subquery for this application_id: 
```
    WITH spar_dev_application_id AS (select application_id from app_fam.fam_application where application_name = 'SPAR_DEV')
```
to reduce the inline query for clarity.
V38_ flyway has example.

Reply:
yeah... I actually followed the v24 where last time we updated the spar roles, also like v39. I'm also fine to update it, both work for me.

Reply:
Sure, can leave it.